apps/web/src/components/Activity/utils.ts

```diff
@@ -0,0 +1,202 @@
+import { IntlShape, MessageDescriptor } from "react-intl";
+import PlusIcon from "@heroicons/react/16/solid/PlusIcon";
+import ArrowPathIcon from "@heroicons/react/16/solid/ArrowPathIcon";
+import TrashIcon from "@heroicons/react/16/solid/TrashIcon";
+import DocumentArrowUpIcon from "@heroicons/react/16/solid/DocumentArrowUpIcon";
+import ExclamationTriangleIcon from "@heroicons/react/16/solid/ExclamationTriangleIcon";
+import { tv, VariantProps } from "tailwind-variants";
+import { isAfter } from "date-fns/isAfter";
+
+import { Maybe, Scalars } from "@gc-digital-talent/graphql";
+import { IconType } from "@gc-digital-talent/ui";
+import { commonMessages } from "@gc-digital-talent/i18n";
+import { parseDateTimeUtc } from "@gc-digital-talent/date-helpers";
+
+import activityMessages from "~/messages/activityMessages";
+
+export const icon = tv({
+  base: "mt-0.5 flex size-5.5 shrink-0 items-center rounded-full bg-primary-500 p-1 text-white",
+  variants: {
+    color: {
+      primary: "bg-primary-500",
+      success: "bg-success-500",
+      warning: "bg-warning-500",
+    },
+  },
+  defaultVariants: {
+    color: "primary",
+  },
+});
+
+type IconVariants = VariantProps<typeof icon>;
+
+type ActivityPropertiesKey = "attributes" | "old";
```

Comment:
If we're very certain on this shape, could the schema type actually provide these?